### PR TITLE
Removing some TODOs 

### PR DESCRIPTION
**Pull request by olegzil**
## Description
Removed the TODOs that could be removed. At present the we need to implement: 
ActivityLifecycleCallbacks -- requires access to the activity
voiceIndex -- variable exists but is not set
bannerIndex -- variable exists but is not set

- [ ] I have added any issue links
- [ ] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

### Goal
cleanup

- [x ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ x] New and existing unit tests pass locally with my changes

## Checklist

- [ x] My code follows the style guidelines of this project
- [ x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt [hunk 1]
```diff
@@ -109,6 +103,9 @@ The class has two public methods, postUserFeedback() and initialize().
 internal object MapboxNavigationTelemetry : MapboxNavigationTelemetryInterface {
     internal const val LOCATION_BUFFER_MAX_SIZE = 20
     private const val ONE_SECOND = 1000
+    private const val MOCK_PROVIDER =
+            "com.mapbox.services.android.navigation.v5.location.replay.ReplayRouteLocationEngine"
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt]:
```suggestion
            "com.mapbox.navigation.core.location.ReplayRouteLocationEngine"
```

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/TelemetryUserFeedback.kt [hunk 2]
```diff
@@ -25,8 +25,7 @@ class TelemetryUserFeedback(
     val locationsAfter: Array<Location>,
     val feedbackId: String,
     val screenshot: String? = null,
-    val step: TelemetryStep? = null,
-    var metadata: TelemetryMetadata
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/TelemetryUserFeedback.kt]: Could we remove `TelemetryUserFeedback` completely (and therefore `TelemetryStep`) and rely only on `NavigationFeedbackEvent`? It sounds like they're duplicated unnecessarily.

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt [hunk 3]
```diff
@@ -657,15 +608,10 @@ internal object MapboxNavigationTelemetry : MapboxNavigationTelemetryInterface {
 
         return SessionState().apply {
             secondsSinceLastReroute = timeSinceReroute
-//             eventRouteProgress: MetricsRouteProgress = MetricsRouteProgress(null)
             eventLocation = newLocation ?: callbackDispatcher.getLastLocation() ?: Location("unknown")
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt]: NIT unnecessary line

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/TelemetryUserFeedback.kt (hunk 2 above)
Comment (olegzil) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/events/TelemetryUserFeedback.kt]: 👍

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt (hunk 3 above)
Comment (olegzil) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetry.kt]: 👍

File: examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt [hunk 4]
```diff
@@ -101,9 +101,9 @@ class SimpleMapboxNavigationKt : AppCompatActivity(), OnMapReadyCallback,
         findViewById<Button>(R.id.btn_send_user_feedback)?.let { button ->
             button.setOnClickListener {
                 MapboxNavigation.postUserFeedback(
-                    TelemetryUserFeedback.FEEDBACK_TYPE_GENERAL_ISSUE,
+                        NavigationFeedbackEvent.FEEDBACK_TYPE_GENERAL_ISSUE,
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt]: NIT unnecessary formatting change

File: examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt [hunk 5]
```diff
@@ -101,9 +101,9 @@ class SimpleMapboxNavigationKt : AppCompatActivity(), OnMapReadyCallback,
...
                     "User feedback test at: ${Date().time}",
-                    TelemetryUserFeedback.FEEDBACK_SOURCE_UI,
+                        NavigationFeedbackEvent.FEEDBACK_SOURCE_UI,
```
Comment (Guardiola31337) [inline examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt]: NIT unnecessary formatting change

File: examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt (hunk 4 above)
Comment (olegzil) [reply examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt]: 👍

File: examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt (hunk 5 above)
Comment (olegzil) [reply examples/src/main/java/com/mapbox/navigation/examples/core/SimpleMapboxNavigationKt.kt]: 👍
